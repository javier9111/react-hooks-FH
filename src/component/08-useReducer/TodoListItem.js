import React from "react";

export const TodoListItem = ({ todo, handleToggle, handleDelete, index }) => {
  return (
    <li key={todo.id} className="list-group-item">
      <p
        onClick={() => handleToggle(todo.id)}
        className={`text-center ${todo.done && "complete"}`}
      >
        {index + 1}.{todo.desc}
      </p>
      <button onClick={() => handleDelete(todo.id)} className="btn btn-danger">
        borrar
      </button>
    </li>
  );
};
