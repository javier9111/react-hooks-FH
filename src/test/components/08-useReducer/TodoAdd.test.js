import React from "react";
import { shallow } from "enzyme";
import { TodoAdd } from "../../../component/08-useReducer/TodoAdd";
describe("Pruebas en <TodoAdd/>", () => {
  const handleAddTodo = jest.fn();
  const wrapper = shallow(<TodoAdd handleAddTodo={handleAddTodo} />);

  test("should show correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should not call handleaddTODO", () => {
    const formSubmit = wrapper.find("form").prop("onSubmit");

    formSubmit({ preventDefault() {} });

    expect(handleAddTodo).toHaveBeenCalledTimes(0);
  });

  test("should call the function handleAddTodo", () => {
    const value = "Aprender React";

    wrapper.find("input").simulate("change", {
      target: { value, name: "description" },
    });
    const formSubmit = wrapper.find("form").prop("onSubmit");

    formSubmit({ preventDefault() {} });
    expect(handleAddTodo).toHaveBeenCalledTimes(1);
    expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object));
    //in order to avoid wich exact value we need to compare to any object we can do the following
    expect(handleAddTodo).toHaveBeenCalledWith({
      id: expect.any(Number),
      desc: value,
      done: false,
    });
    //this evaluation check if the method reset() has been executed.
    expect(wrapper.find("input").prop("value")).toBe("");
  });
});
