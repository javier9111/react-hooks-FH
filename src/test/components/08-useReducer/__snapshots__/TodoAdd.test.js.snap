// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoAdd/> should show correctly 1`] = `
<Fragment>
  <h4>
    Agregar to Do
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      name="description"
      onChange={[Function]}
      placeholder="aprender ..."
      type="text"
      value=""
    />
    <div
      className="d-grid gap-2 mt-1"
    >
      <button
        className="btn btn-primary"
        type="submit"
      >
        agregar
      </button>
    </div>
  </form>
</Fragment>
`;
